Add Dockerfile.development for quick development

tristil opened the pull request:
The idea is that if you just want to make a quick fix and run the specs
to make sure you didn't break something you can just run
`docker-compose run dev bash` to drop into an environment that can
compile Crystal.

(This ended up being the path of least resistance because of various issues with compiling on either MacOS or Ubuntu, so I thought this might be useful for others.)

waterlink commented:
I am using this script `./bin/in-docker`:

```bash
#!/usr/bin/env bash

[[ -z "$ARCH" ]] && ARCH=$(arch)
docker run \
    -it \
    --rm \
    -u $(id -u) \
    -v $(pwd):/mnt \
    -w /mnt \
    -e LIBRARY_PATH="/opt/crystal/embedded/lib/" \
    "jhass/crystal-build-$ARCH" \
    "$@"
```

It re-uses existing docker image that is used for building crystal at CI. Works very well on Linux (probably will need to have some tweaks for usage on `docker-machine`).

tristil commented:
@waterlink I updated to just use that image in my Dockerfile and dropped all the duplicate dependencies. Your script would be okay to use instead of what I have but how do you get libpcl into `/opt/crystal/embedded` ? I don't see libpcl here: https://hub.docker.com/r/jhass/crystal-build-x86_64/~/dockerfile/ 

I like the idea of just having one executable to get started, rather than having to know about `docker-compose`, something like `./bin/development-quick-start` maybe.

jhass commented:
`libpcl` is shipped with the packages: https://github.com/manastech/omnibus-crystal/blob/master/config/software/libpcl.rb

tristil commented:
@jhass But don't I need it if I'm compiling a new Crystal inside the docker image?

jhass commented:
You do, hence it's shipped. No need to download separately.

tristil commented:
I'm going to close this and tinker around with it.
